pkg/runtime/reconciler.go

```diff
@@ -376,10 +383,30 @@ func (r *resourceReconciler) createResource(
 		return latest, err
 	}
 
-	rlog.Enter("rm.ReadOne")
-	observed, err := rm.ReadOne(ctx, latest)
-	rlog.Exit("rm.ReadOne", err)
+	bo := backoff.NewExponentialBackOff()
+	bo.MaxElapsedTime = backoffReadOneTimeout
+	ticker := backoff.NewTicker(bo)
+	attempts := 0
+
+	var observed acktypes.AWSResource
+
+	for range ticker.C {
+		rlog.Enter(fmt.Sprintf("rm.ReadOne (attempt %d)", attempts+1))
+		observed, err = rm.ReadOne(ctx, latest)
+		rlog.Exit(fmt.Sprintf("rm.ReadOne (attempt %d)", attempts+1), err)
+		if err != nil {
+			if attempts+1 > backoffReadOneMaxRetries {
+				ticker.Stop()
+				break
+			}
+			continue
+		}
+		ticker.Stop()
+		break
```

Comment:
We need to handle non-404 errors differently than 404s. Remove the max retries stuff and update this block to this instead:

```go
if err == nil {
    break
}
if err != ackerr.NotFound {
    return latest, nil
}
```

Reply:
1) Place all of the retry w/ backoff loop into a separate helper function called `delayedReadOneAfterCreate()`.

2) Use the `rlog.Trace()` functionality to mark the entrance and exit to this helper function.

3) The `delayedReadOneAfterCreate` helper function represents an exceptional edge case and as such we should only call the `delayedReadOneAfterCreate` helper function if the original code's `ReadOne` call returns an `err` that is not `nil` and is a NotFound.

In other words, the original code was this:

```go
	rlog.Enter("rm.Create")
	latest, err = rm.Create(ctx, desired)
	rlog.Exit("rm.Create", err)
	if err != nil {
		return latest, err
	}

	rlog.Enter("rm.ReadOne")
	observed, err := rm.ReadOne(ctx, latest)
	rlog.Exit("rm.ReadOne", err)
	if err != nil {
		return latest, err
	}
```

and I'd like to see it look like this:

```go
	rlog.Enter("rm.Create")
	latest, err = rm.Create(ctx, desired)
	rlog.Exit("rm.Create", err)
	if err != nil {
		return latest, err
	}

	rlog.Enter("rm.ReadOne")
	observed, err := rm.ReadOne(ctx, latest)
	rlog.Exit("rm.ReadOne", err)
	if err != nil {
		if err == ackerr.NotFound {
			// Some eventually-consistent APIs return a 404 from a
			// ReadOne operation immediately after a successful
			// Create operation. In these exceptional cases
			// we retry the ReadOne operation with a backoff
			// until we get the expected 200 from the ReadOne.
			observed, err = r.delayedReadOneAfterCreate(ctx, latest)
			if err != nil {
				return latest, err
			}
		} else {
			return latest, err
		}
	}
```

Reply:
In the case of a server 5xx error, do we want to immediately return that? I could imagine a hiccup in the service (or maybe a throttling exception) could cause our backoff to false negative and spit out an error?

Reply:
I like the idea of the `delayedReadOneAfterCreate`. Changed the code to adopt that.

Reply:
> In the case of a server 5xx error, do we want to immediately return that? I could imagine a hiccup in the service (or maybe a throttling exception) could cause our backoff to false negative and spit out an error?

I think it's really hard to generalize the code to be able to handle 5XX errors across all the AWS APIs. Also 5xx errors can occur at any CRUD operation